### ISSUE TITLE
Skipped tests can't set up testMetadata types

### ISSUE THREAD

**Issue by choheekim** [emberjs/ember-test-helpers#748]
QUnit assign skipped tests callbacks to be an empty function so that the tests are completely ignored.(https://github.com/qunitjs/qunit/blob/5bea87b0f2974b49e8bf907aa2b197ee469fe99b/src/test.js#L83)

As all hooked functions (e.g. `setupApplicationContext`, `setupRenderingContext`) are not invoked skipped tests don't set up types and end up neither application nor rendering tests.

**Comment by rwjblue** [comment]
> As all hooked functions (e.g. setupApplicationContext, setupRenderingContext) are not invoked skipped tests don't set up types and end up neither application nor rendering tests.

Yep, I definitely understand this, but I don't particularly consider this a bug. Are you saying that you believe that it is? That we need to update our documentation to explain it? Something else?

I'm a bit confused by this issue at the moment. 🤔

**Comment by choheekim** [comment]
@rwjblue I wasn't really sure if it's an expected behavior. I believed that testMetadata should be available for all tests whether or not it's skipped.

**Comment by rwjblue** [comment]
It would only work for tests that are ran (skipped or filtered tests would not have metadata).
